entrypoint.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+fail () {
+    echo "$@" >&2
+    exit 1
+}
+
+shopt -s nullglob
+cvmfs_mounts=(/cvmfs/*)
+if [[ -z $cvmfs_mounts && -n $CVMFSEXEC_REPOS && -n $CVMFSEXEC_DIST ]]; then
+    echo "No cvmfs mounts; using cvmfsexec to mount $CVMFSEXEC_REPOS"
+    # We need our own copy of cvmfsexec for permissions reasons.
+    cvmfsexec_root=/tmp/cvmfsexec-$(id -u)
+    if [[ ! -d $cvmfsexec_root ]]; then
+        echo "No cvmfsexec dir found for this user at $cvmfsexec_root; creating one"
+        cp -rp /cvmfsexec $cvmfsexec_root || fail "Couldn't create $cvmfsexec_root"
```

Comment:
The copy in /tmp is owned by the user that's running the container, which may not be the user that created the container.

Reply:
Maybe chown -R would be lighter weight.

Reply:
Except that probably this script isn't running as fake root so chown wouldn't work.

Reply:
Exactly. I create a "template" in /cvmfsexec which each user (as themselves) copies to create their own cvmfsexec directory into which they have write permissions.

I wonder if it would be a good idea to run makedist at image creation, tar up the results and, at startup time have, have the user untar the results to create their own copy...

Reply:
That's better than downloading at start time, but has the same issues as makedist -o I brought up in the [related thread on this ticket](https://github.com/opensciencegrid/osgvo-docker-pilot/pull/9#discussion_r599981246).  Follow up there.